src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -267,6 +275,17 @@ class FastSync(
           )
           syncState =
             syncState.updatePivotBlock(pivotBlockHeader, syncConfig.fastSyncBlockValidationX, updateFailures = true)
+
+        case NodeRestart =>
```

Comment:
Shouldn't it be named `SyncRestart`? If fast-sync actor gets restarted due to some failure catched by supervisor, it's going to be restarted with clean state the same way is if it was restarted whole node. 

That also makes me think - shouldn't `SyncController` watch for Fast-Sync restarts and start it once such happens?

Reply:
So i agree that `SyncRestart` is more compelling name.

Question about supervision is more nuanced, in general I am not sure we handle it well across all codebase, but for this particual case we are fine as: `FastySync` is child of `SyncController` and default strategy for uncatched `Exception` in child is just restart it.  It will probably mean some of the request in flight will be later ignored and some of the peers gets unnecessry blacklis. Those missed requests may triger some weird error condition. In my view this whole class was not designed with restarts in mind, but rather with handing all excpetion by itself.